src/Rpc/NodeGuardService.cs

```diff
@@ -868,6 +870,42 @@ private bool ValidateTargetBalance(LiquidityRule rule)
         return true;
     }
 
+    public override async Task<GetAllUtxosResponse> GetAllUtxos(GetAllUtxosRequest request, ServerCallContext context)
+    {
+        List<FMUTXO> fmutxos;
+        // Check if the user sent a wallet id, else we get all utxos
+        if (request.WalletId == 0)
+        {
+            fmutxos = await _fmutxoRepository.GetAll();
+        }
+        else
+        {
+            fmutxos = await _fmutxoRepository.GetByWalletId(request.WalletId);
+        }
+
+        // Check if user wants the list to contain the locked utxos
+        if (!request.IncludeLocked)
+        {
+            var lockedUtxos = await _fmutxoRepository.GetLockedUTXOs();
+            Console.WriteLine(lockedUtxos.Count);
+            Console.WriteLine(fmutxos.Count);
+            fmutxos = fmutxos.Except(lockedUtxos).ToList();
+        }
+
+        var utxos = fmutxos.Select(utxo => new Utxo()
+        {
+            Amount = new Money(utxo.SatsAmount),
+            // We follow the same convention as GetAvailableUtxos for displaying outpoints
+            Outpoint = $"{utxo.TxId}-{utxo.OutputIndex}",
+            Address = utxo.Address?.ToString()
```

Comment:
can you explain to me why an address could be null?

Reply:
it shouldn't. I think it is nullable because it's a new row and it is not filled by default. @markettes where do we go back and fill this for utxos already on the table?

Reply:
We don't really need to fill these old UTXOs, we could just remove the data from this table and it will be generated.

Reply:
cool